### DISCUSSION TITLE
How loading from Git LFS (Large File Storage) might work in Next.js on Vercel?

### DISCUSSION THREAD

**Discussion by lancejpollard** [vercel/next.js#39793] [Help] [answered]
I am considering building an app which uses Git LFS (which costs about $5 per 50GB on GitHub) to store let's say a terabyte of images/audios/videos/etc. (which would be 20 50GB packs so $100/mo, or at $0.023/GB on S3, about $23 there). In addition to the terabyte of "large files", the repo itself might grow up to several Gigabytes (let's say 100GB, the max GitHub allows).

If the 100GB of "JSON data" is in the form of JSON, YAML, CSV, and JavaScript files, and gets compiled to memory at runtime into a big Map or something of JSON objects, that would take a long time to "load" all that structured data into JavaScript objects, probably several minutes if not longer.

So I am wondering how a deploy would work/look in this scenario on Vercel with Next.js. Would a push to the repo download the diff of all the terabyte of binary files, plus the diff of the 100GB of structured data, and then load it up somehow, and only after it is loaded flip a switch to the new live app? Or how does it work (or is it even possible).

The 100GB of structured data in theory would be then rendered into the Next.js app, such as through the /api folder, but just reading in-memory JSON structures.

I guess so far, first, 100GB won't fit into the 1GB memory max allowed on Vercel.

**Comment by gs0510** [comment, marked as answer]
Hi @lancejpollard, [we now support LFS in Vercel](https://vercel.com/changelog/git-lfs-support).

> Would a push to the repo download the diff of all the terabyte of binary files, plus the diff of the 100GB of structured data, and then load it up somehow, and only after it is loaded flip a switch to the new live app?
Vercel will fetch the LFS objects of the current branch, so it does download the objects. We also cache the repo so only the new objects would be fetched.

> I guess so far, first, 100GB won't fit into the 1GB memory max allowed on Vercel.
Yes, the 1GB limit is on the compressed build cache, the 100 GB would not fit the 1 GB memory max on Vercel.

Hope this answers the questions! If you have any more questions, happy to answer them!

**Reply by lancejpollard** [reply]
Can you state real quick here if it downloads everything on every build, or not? Is it optimized for large files basically? Does it just get the diff in some optimized way? That link is quite intense, but I will read through it soon enough :)
